pkg/perf/cache.go

```diff
@@ -0,0 +1,43 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// TaskResult the result test schema
+type TaskResult struct {
+	TaskName     string      `json:"task_name"`
+	RunTimestamp string      `json:"run_timestamp"`
+	Result       interface{} `json:"result"`
+}
+
+// setCache set result in redis
+func (pm *PerformanceMonitor) setCache(ctx context.Context, taskName string, result TaskResult) error {
+	data, err := json.Marshal(result)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal data to JSON")
+	}
+
+	_, err = pm.redisConn.Do("SET", taskName, data)
```

Comment:
Using the task name alone as a key is not a good idea. The same redis instance is used by other apps on the system to store and queue data. Hence you need to create some sort of a `namespacing` for your data. for example prefix all keys with `perf.` so for example a `cpu` test key will be `perf.cpu` instead of `cpu` so it has less chance of having collisions with other apps

Having a prefix also makes it easier for range scan. For example if u wanna return ALL tests u can scan KEYS with `perf.*` to return all results for all tests